### DISCUSSION TITLE
How do you mix on dawdreamer?

### DISCUSSION THREAD

**Discussion by samuelbraun04** [DBraun/DawDreamer#136] [Q&A] [answered]
For example, how could I a mix a .wav file to -6 dB?

**Comment by DBraun** [comment, marked as answer]
You could find the decibels of the input signal in advance. See https://stackoverflow.com/questions/63347977/what-is-the-conceptual-purpose-of-librosa-amplitude-to-db

Then suppose you want to apply an additional decibel reduction on it (-6dB). This is some [Faust code](https://faustide.grame.fr/?autorun=1&voices=0&name=untitled&inline=aW1wb3J0KCJzdGRmYXVzdC5saWIiKTsKZ2FpbiA9IGhzbGlkZXIoImxvdWRuZXNzIiwgMC4sIC0xMDAsIDIwLiwgLjAwMSkgOiBiYS5kYjJsaW5lYXI7CnByb2Nlc3MgPSBwYXIoaSwgMiwgXypnYWluKTs%3D) for that

```faust
import("stdfaust.lib");
gain = hslider("loudness", -6., -100, 20., .001) : ba.db2linear;
process = par(i, 2, _*gain);
```
python stuff:
```bash
>>> import dawdreamer
>>> e = dawdreamer.RenderEngine(44100, 1)
>>> f = e.make_faust_processor("faust")
>>> c = """import("stdfaust.lib");gain = hslider("loudness", -6., -100, 20., .001) : ba.db2linear;process = par(i, 2, _*gain);"""
>>> assert f.set_dsp_string(c)
>>> f.get_parameters_description()
[{'index': 0, 'name': '/dawdreamer/loudness', 'numSteps': 120000, 'isDiscrete': False, 'label': 'loudness', 'min': -100.0, 'max': 20.0, 'step': 0.0010000000474974513, 'value': -6.0}]
>>>
```

Then you would connect your audio to that Faust Processor. You can also set the loudness anytime:
```f.set_parameter("/dawdreamer/loudness", -12)```
